impl/src/main/java/org/jboss/weld/util/Observers.java

```diff
@@ -84,7 +85,8 @@ private Observers() {
 
     public static boolean isContainerLifecycleObserverMethod(ObserverMethod<?> method) {
         return CONTAINER_LIFECYCLE_EVENT_TYPES.contains(Reflections.getRawType(method.getObservedType()))
-                || (Object.class.equals(method.getObservedType()) && method instanceof ContainerLifecycleEventObserverMethod);
+                || (Object.class.equals(method.getObservedType()) && method.getObservedQualifiers().size() == 1
```

Comment:
`void observe(@Observes Object event, Foo foo)` has no observed qualifiers so it would NOT be recognized as a container lifecycle event - that's not correct. Also `org.jboss.weld.tests.extensions.validation.WildcardObserverTest` accidentally passes (the deployment fails with "_org.jboss.weld.exceptions.IllegalArgumentException: WELD-001456: Argument resolvedBean must not be null_").